### PR TITLE
Support Ruby 2.4.1

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR **adds Ruby 2.4.1** to the Travis CI matrix.

To pass tests on 2.4, it avoids the Fixnum class.

Also: all EOL'd Ruby versions are now in the `allow_failures` list. [Blog post on 2.1 being EOL](https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/).

### CODE REVIEW

**Comment by olleolleolle** [comment]
@laserlemon There.

File: .travis.yml [hunk 1]
```diff
@@ -11,9 +11,12 @@ matrix:
...
+    - rvm: "2.1"
 rvm:
   - "2.0"
   - "2.1"
-  - "2.2"
+  - "2.2.7"
   - "2.3.4"
+  - "2.4.1"
   - ruby-head
+
```
Comment (laserlemon) [inline .travis.yml]: Unnecessary whitespace. 😜

File: spec/json_spec/matchers/have_json_type_spec.rb [hunk 2]
```diff
@@ -56,8 +56,14 @@
...
-    matcher.failure_message_when_negated.should   eq "Expected JSON value type to not be Numeric, got Fixnum"
-    matcher.failure_message_for_should_not.should eq "Expected JSON value type to not be Numeric, got Fixnum" # RSpec 2 interface
+    klass = if 1.class == Integer
+      'Integer'
+    else
+      'Fixnum'
+    end
+
+    matcher.failure_message_when_negated.should   eq "Expected JSON value type to not be Numeric, got #{klass}"
+    matcher.failure_message_for_should_not.should eq "Expected JSON value type to not be Numeric, got #{klass}" # RSpec 2 interface
```
Comment (laserlemon) [inline spec/json_spec/matchers/have_json_type_spec.rb]: Rather than the fork, maybe we can change line 58 to use a float instead and just tweak the assertion. The intent of this test would still be asserted. Thoughts?

**Comment by olleolleolle** [comment]
@laserlemon There, avoids the Fixnum class.

**Comment by laserlemon** [comment]
👏
